migrations/1667412728068-DepositReferralStats.ts

```diff
@@ -0,0 +1,114 @@
+import { MigrationInterface, QueryRunner } from "typeorm";
+
+export class DepositReferralStats1667412728068 implements MigrationInterface {
+  name = "DepositReferralStats1667412728068";
+
+  public async up(queryRunner: QueryRunner): Promise<void> {
+    await queryRunner.query(
+      `ALTER TABLE "merkle_distributor_recipient" DROP CONSTRAINT "UK_merkle_distributor_recipient_merkleDistributorWindowId_addre"`,
+    );
+    await queryRunner.query(
+      `ALTER TABLE "merkle_distributor_recipient" ADD CONSTRAINT "UK_merkle_distributor_recipient_merkleDistributorWindowId_address" UNIQUE ("merkleDistributorWindowId", "address")`,
+    );
+
+    await queryRunner.query(`drop materialized view if exists deposits_mv`);
+    await queryRunner.query(`DROP VIEW IF EXISTS "deposit_referral_stats";`);
+
+    await queryRunner.query(`CREATE VIEW "deposit_referral_stats" AS 
+    SELECT d1.id,
+      count(distinct d2."depositorAddr") as "referralCount",
+      sum((d2.amount / power(10, t.decimals)) * hmp.usd) as "referralVolume"
+    FROM deposit d1
+    LEFT JOIN deposit d2 on d1."stickyReferralAddress" = d2."stickyReferralAddress"
+      AND d1."depositDate" >= d2."depositDate"
+      AND d1."stickyReferralAddress" is not null
+      AND d1."depositDate" is not null
+      AND d1."tokenId" is not null
+      AND d1."priceId" is not null
+      AND d1.status = 'filled'
+      AND d2."depositDate" >= (SELECT  coalesce(max(c."claimedAt"), '2000-01-01 00:00:00')
+                               FROM    claim c
+                               WHERE   c.account = d2."stickyReferralAddress"
+                               AND     c."windowIndex" >= 1)
```

Comment:
Hmm, I might be wrong, but this should not be the time of the claimed event, but the timestamp when the referral rewards were computed for the airdrop, right?

Reply:
Yes, you are right. So I need to link the `MerkleDistributorWindow` entity here, I assume. But which timestamp do we consider? 🤔 I think we need to add additional payload to the tree metadata for referral rewards, something like `snapshotAt` and use this.

Reply:
When we do these referral reward snapshots, we need to make sure to somehow only include deposits with a high number of confirmations to prevent double rewards for re-orged deposits.